pkg/install/check/check.go

```diff
@@ -0,0 +1,525 @@
+package check
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/acorn-io/acorn/pkg/client/term"
+	"github.com/acorn-io/acorn/pkg/k8schannel"
+	"github.com/acorn-io/acorn/pkg/k8sclient"
+	"github.com/acorn-io/acorn/pkg/scheme"
+	"github.com/acorn-io/acorn/pkg/streams"
+	"github.com/acorn-io/acorn/pkg/system"
+	"github.com/acorn-io/baaah/pkg/restconfig"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CheckResult describes the results of a check, making it human-readable
+type CheckResult struct {
+	Message string `json:"message"`
+	Passed  bool   `json:"passed"`
+	Name    string `json:"name"`
+}
+
+// PreflightChecks is a list of all checks that are run before the installation.
+// They are crictial and will make the installation fail.
+func PreflightChecks(ctx context.Context) []CheckResult {
+	return RunChecks(ctx, CheckRBAC, CheckNodesReady)
+}
+
+// InFlightChecks is a list of all checks that are run after the installation.
+// They are not critical and should not affect the installation process.
+func InFlightChecks(ctx context.Context) []CheckResult {
+	checks := []func(ctx context.Context) CheckResult{
+		CheckDefaultStorageClass,
+		CheckIngressCapability,
+		CheckExec,
+	}
+
+	// Some debugging test
+	if os.Getenv("ACORN_INSTALL_FAIL_CHECKS") == "true" {
```

Comment:
wonder if we should keep this

Reply:
It was actually only for you two to test this. For some of this I found it cumbersome to come up with scenarios where the checks failed. I will delete it :)